src/Xamarin.Android.Build.Tasks/Xamarin.Android.CSharp.targets

```diff
@@ -42,7 +42,7 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
         <DebugSymbols Condition=" '$(DebugType)' == 'None' ">true</DebugSymbols>
         <DebugType Condition=" '$(DebugType)' == 'None' Or '$(DebugType)' == '' ">portable</DebugType>
     </PropertyGroup>
-    <Import Project="Xamarin.Android.DefaultOutputPaths.targets" Condition="'$(IsXBuild)' != 'true' and '$(EnableDefaultOutputPaths)' == 'true'" />
+    <Import Project="Xamarin.Android.DefaultOutputPaths.targets" Condition=" '$(OS)' == 'Windows_NT' And '$(EnableDefaultOutputPaths)' == 'true'" />
```

Comment:
I don't like this change because it'll complicate Visual Studio for Mac's efforts in testing the new `Xamarin.Android.DefaultOutputPaths.targets` codepath/support. Rephrased: how does anybody on macOS use `Xamarin.Android.DefaultOutputPaths.targets` with this change in place?  (They don't; they'd have to manually edit this file for testing, which seems ugly.)

I think a "saner" approach might be to move this to `$(EnableDefaultOutputPaths)`:

```xml
<EnableDefaultOutputPaths Condition=" '$(EnableDefaultOutputPaths)' == '' And '$(OS)' != 'Windows_NT' ">False</EnableDefaultOutputPaths>
<EnableDefaultOutputPaths Condition="'$(EnableDefaultOutputPaths)' == ''">True</EnableDefaultOutputPaths>
```

This would allow macOS developers to enable the new system by explicitly setting `/p:EnableDefaultOutputPaths=True`.

Reply:
We also need to stop the code messing with the `BaseIntermediateoutputPath` because it seems that is what is causing the problems